### PR TITLE
feat(catalog/equipment): add equipment templates reference catalogue (#708)

### PR DESCRIPTION
**Pull request by benoit-bremaud**
## Summary

Phase 3 PR #7 of the brewing reference catalogues refactor (Issue #708 / #869). Adds the **EquipmentCatalogModule** under the existing top-level \`catalog/\` module, seeded with **9 equipment template entries** spanning the full beginner-to-pro spectrum — from a 5L kitchen pot for first-time brewers up to a 23L 3-Vessel HERMS for advanced setups.

## Naming: \`equipment_templates\` (not \`equipment_profiles\`)

The table is called \`equipment_templates\` and the class \`EquipmentTemplateOrmEntity\` to avoid the class-name collision with the existing top-level \`src/equipment/equipment.module.ts:EquipmentModule\` and its \`EquipmentProfileOrmEntity\` (user-owned per-brewer setups). A template here is the SHARED reference (manufacturer specs) that an end user can later copy to seed their own personal \`equipment_profile\`.

This is the same naming pattern adopted on PR #894 for \`WaterCatalogModule\` (after Copilot caught the same kind of collision risk).

| Existing user-owned | New catalogue (this PR) |
|---|---|
| \`equipment_profiles\` table | \`equipment_templates\` table |
| \`EquipmentProfileOrmEntity\` | \`EquipmentTemplateOrmEntity\` |
| \`EquipmentModule\` | \`EquipmentCatalogModule\` |
| Per-user, mutable | Operator-curated, immutable |
| Referenced by user batches | Used as starting template for personal profiles |

## Schema (table \`equipment_templates\` — 18 columns total: 15 data + 3 system)

15 data columns map BeerXML 1.0 \`<EQUIPMENT>\` field-by-field:
- \`name\` UNIQUE
- Volumes (L): \`boil_size_l\`, \`batch_size_l\`, \`tun_volume_l\`, \`top_up_water_l\`, \`trub_chiller_loss_l\`, \`lauter_deadspace_l\`, \`top_up_kettle_l\`
- Thermal: \`tun_weight_kg\`, \`tun_specific_heat\`
- Process: \`evap_rate_percent\`, \`boil_time_min\`, \`calc_boil_volume\`, \`hop_utilization_percent\`
- \`notes\` (FR, UI-facing)

Indexes: UNIQUE on \`name\`.

## Endpoints

\`\`\`
GET /catalog/equipment-templates           → ordered list
GET /catalog/equipment-templates/:uuid     → single entry, 404 on miss
\`\`\`

## Seed (9 entries, idempotent loader)

Operator-curated, drawn from the BeerXML reference fixture and manufacturer datasheets, structured to give every beginner an immediate match.

- **1 Brasse-Bouillon original kitchen starter**: Casserole cuisine 5L (kit d'initiation extract — true beginner who only owns kitchen pots; 5L final batch ~6 bottles, no investment required)
- **2 BeerXML canonical** (verbatim from \`libraries/equipment.xml\`): Brew Pot 4 Gal (extract), Brew Pot 6+gal + Igloo Cooler 5 Gal (all-grain classic)
- **6 modern popular setups**: BIAB 20L, BIAB 30L, Grainfather G30 (electric all-in-one), Klarstein Brauheld Pro 30L (EU entry-level), Anvil Foundry 6.5 Gal (US popular electric), 3-Vessel HERMS Recirculating (advanced / pro)

Coverage: **batch sizes 5L → 23L**, full beginner-to-pro spectrum.

The kitchen starter explicitly addresses the \"I just want to try brewing once with what I already own\" persona — no investment required, immediately usable on a kitchen stove.

## Tests (10 new, all green)

- **Service spec** — in-memory SQLite (4 tests): list ordered alphabetically, getById happy / sad / edge (UUID strict)
- **Controller spec** — \`jest.spyOn\` pattern (4 tests)
- **Seed spec** — delegates to \`assertCommonCatalogueSeederBehaviours\` helper (4 standard scenarios) + 5 catalogue-specific assertions (9-entry name coverage, deterministic UUID range \`…-9000-6…\`, batch-size spectrum 5L-23L, French notes invariant, basic field type checks)

## All six review lessons baked in (PR #888 + PR #894)

- ✅ **Boot-safe registration** — \`EquipmentTemplateOrmEntity\` added to \`ormEntities\`
- ✅ **Migration consistency** — \`CREATE UNIQUE INDEX IDX_equipment_templates_name\` (matches IDX_styles_name etc.)
- ✅ **No filtered query params** so no ParseEnumPipe needed
- ✅ **Runner script** — \`scripts/run-equipment-templates-catalog-seed.ts\`
- ✅ **Entity \`unique: true\`** on the name column (matches the migration UNIQUE INDEX — no test/prod drift under \`TYPEORM_SYNCHRONIZE=true\`)
- ✅ **\`Catalog\` suffix on module class name** to avoid collision with existing top-level \`EquipmentModule\` (lesson from PR #894 \`WaterCatalogModule\`)
- ✅ **Consistent NotFoundException message** — \"Equipment template catalogue entry \${id} not found\" (matches Hop / Yeast / Style / Fermentable / Mash / Water format)

## Phase 3 status after this PR

| # | PR | Catalogue | Status |
|---|---|---|---|
| 6 | #894 | waters | ✅ merged |
| 7 | this PR | equipment_templates | 🟡 awaiting review |
| 8 | next | misc | pending |

After this PR merges, only one catalogue remains (MISC for spices / finings / water agents) before the post-Phase 1-3 normalize-producers PR.

## Checklist

- [x] Happy / sad / edge cases covered (10 new tests)
- [x] \`npm run lint:check\` passes (0 errors)
- [x] \`npm run build\` passes
- [x] All existing tests still green (545 → 564 total — 9 new seed entries + 10 new test cases)
- [x] No \`any\` introduced
- [x] Migration timestamp picks up where the previous one left off (\`1789000000000\`)
- [x] All six prior review lessons baked in upfront

Refs #708 #869

### CODE REVIEW

**Comment by benoit-bremaud** [comment]
@vitalikevin @Smith06S @Thais9723 — FYI : avant-dernière PR catalogues — les **modèles de matériel** brassicole.

Cette PR ajoute la **table des templates de matériel** : 9 setups types qui couvrent toute la palette du brasseur, du grand débutant au semi-pro. Particularité importante : on a explicitement intégré un profil "casserole de cuisine 5L" pour le user qui veut juste essayer le brassage **sans rien acheter** — il prend ce qu'il a déjà dans sa cuisine, fait 5L de bière (~6 bouteilles 75 cL) à partir d'un kit d'extrait de malt. C'est l'entrée la plus accessible possible.

Le reste des 9 entrées : la casserole 4 gallons américaine standard, le BIAB 20/30L (méthode "sac" la plus simple en all-grain), les 3 systèmes électriques tout-en-un les plus populaires (Grainfather, Klarstein, Anvil), et un setup HERMS pro 3 cuves pour les avancés.

À quoi ça sert pour la démo : quand Lucas s'inscrit, il choisit son matériel parmi ces 9 templates → l'app crée pour lui un profil personnel pré-rempli avec les bonnes valeurs (volume cuve, taux d'évaporation, pertes au refroidisseur, etc.). Plus de chiffres à deviner. Ses brassins futurs utiliseront ce profil personnel pour calculer précisément les volumes d'eau, les apports en houblons, etc.

**Important pour distinguer du existant** : il existe déjà une table `equipment_profiles` pour le matériel personnel des utilisateurs (modifiable par eux). Cette PR crée une table SÉPARÉE `equipment_templates` pour les modèles de référence (immuables, partagés). Sémantique : 'template = modèle à copier'. Module nommé `EquipmentCatalogModule` (avec le suffix Catalog comme `WaterCatalogModule` de la PR précédente, pour éviter toute collision de nom).

C'est la **PR #7/8** des catalogues. Reste juste **MISC** (épices, finings, agents d'eau) pour boucler la Phase 3, puis la PR transversale de normalisation des producteurs.

File: packages/api/src/catalog/equipment/services/equipment-catalog.service.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    return this.templates.find({
+      order: { name: 'ASC' },
+    });
+  }
+
+  /**
+   * Returns a single catalogue entry by its UUID, or throws 404.
+   * Strict UUID match — no name lookup here, since variants of
+   * the same hardware (e.g. "Grainfather G30 v3" vs "Grainfather
+   * G30 v4") may share the same display name across revisions.
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/catalog/equipment/services/equipment-catalog.service.ts]: Good catch — the justification was misleading. The schema enforces `UNIQUE(name)` (see migration line 45 + entity line 48), so multiple revisions could not share a display name even if we wanted them to. I rewrote the docstring to lean on the actual reasons we keep this UUID-only: (1) consistency with every other catalogue service (Hop / Yeast / Style / Fermentable / Mash / Water), and (2) keeping the API surface minimal — the picker UI already carries the UUID once a user has selected an entry, so a name-lookup endpoint would be added when there is a real use case for it. Fixed in 2b42d02.

File: packages/api/src/catalog/equipment/entities/equipment-template.orm.entity.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  UpdateDateColumn,
+} from 'typeorm';
+
+/**
+ * Immutable reference catalogue of homebrew equipment templates
+ * (Issue #708 / #869, Phase 3 PR #7). Seeded with 8 entries —
+ * the 2 BeerXML 1.0 canonical equipment profiles (verbatim from
+ * `libraries/equipment.xml`) plus 6 popular modern setups
+ * (BIAB 20/30L, Grainfather G30, Klarstein Brauheld Pro,
+ * Anvil Foundry, 3-Vessel HERMS).
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/catalog/equipment/entities/equipment-template.orm.entity.ts]: Agreed — stale doc from when the kitchen-starter template was added during review. Bumped "8" → "9" and rewrote the parenthetical to call out the three groups explicitly: 1 Brasse-Bouillon original kitchen starter (5L casserole, extract brewing, no investment), 2 BeerXML 1.0 canonical equipment profiles (verbatim from libraries/equipment.xml), and 6 popular modern setups (BIAB 20/30L, Grainfather G30, Klarstein Brauheld Pro, Anvil Foundry, 3-Vessel HERMS). Fixed in 2b42d02.

File: packages/api/src/database/migrations/1789000000000-AddEquipmentTemplatesCatalog.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ *
+ * **Naming**: distinct from the existing `equipment_profiles`
+ * table (user-owned per-brewer setups). A template here is the
+ * SHARED reference (manufacturer specs) that a user can later
+ * copy to seed their own personal `equipment_profile`. The
+ * naming split avoids the kind of class-name collision that
+ * PR #894 caught and fixed for the water catalogue.
+ *
+ * Seeded with 8 entries (2 BeerXML verbatim from
+ * libraries/equipment.xml + 6 popular modern setups).
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/database/migrations/1789000000000-AddEquipmentTemplatesCatalog.ts]: Good catch. Migration headers are immutable in production once shipped (rewriting them would change the migration's effective payload and risk re-running it on existing DBs), but this migration has not been deployed anywhere yet — it is part of the same unmerged PR that adds the 9th template — so updating the header here is safe and keeps the audit trail honest. Bumped to "9 entries (1 Brasse-Bouillon original kitchen starter + 2 BeerXML verbatim from libraries/equipment.xml + 6 popular modern setups)" in 2b42d02.

File: packages/api/scripts/run-equipment-templates-catalog-seed.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+/**
+ * One-shot script to seed the `equipment_templates` reference
+ * catalogue with the 8 curated entries from
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/scripts/run-equipment-templates-catalog-seed.ts]: Same root cause as the other three doc comments — stale reference from before the kitchen-starter template was added. Bumped the runner header to "9 curated entries" in 2b42d02 so the operator running this script sees the right count vs the actual seed payload.
